### PR TITLE
Play upgrade

### PR DESCRIPTION
**Pull request by jeremystone**

### CODE REVIEW

File: app/uk/gov/hmrc/r2/selfassessmentapi/config/AppContext.scala [hunk 1]
```diff
@@ -17,32 +17,31 @@
...
-  lazy val mtdDate: String = config.getString(s"$env.mtd-date").getOrElse(throw new RuntimeException("mtd-date is not configured"))
-
-  override protected def mode: Mode = environment.mode
-
-  override protected def runModeConfiguration: Configuration = config
+  lazy val apiStatus: String = servicesConfig.getString("api.status")
+  lazy val desUrl: String = servicesConfig.baseUrl("des")
+  lazy val featureSwitch: Option[Configuration] = config.getOptional[Configuration](s"${runMode.env}.feature-switch")
+  lazy val auditEnabled: Boolean = config.getOptional[Boolean](s"auditing.enabled").getOrElse(true)
+  lazy val authEnabled: Boolean = config.getOptional[Boolean](s"${runMode.env}.microservice.services.auth.enabled").getOrElse(true)
```
Comment (narendravijayarao) [inline app/uk/gov/hmrc/r2/selfassessmentapi/config/AppContext.scala]: Everything is fine but worried about this getOrElse(true)... we can do them later also... better check with Mark

File: app/uk/gov/hmrc/r2/selfassessmentapi/config/FeatureSwitch.scala [hunk 2]
```diff
@@ -38,21 +38,21 @@ case class FeatureSwitch(value: Option[Configuration], env: String) {
 
   def isWhiteListingEnabled: Boolean = {
     value match {
-      case Some(config) => config.getBoolean("white-list.enabled").getOrElse(false)
+      case Some(config) => config.getOptional[Boolean]("white-list.enabled").getOrElse(false)
       case None => false
     }
   }
 
   def isAgentSimulationFilterEnabled: Boolean = value match {
```
Comment (narendravijayarao) [inline app/uk/gov/hmrc/r2/selfassessmentapi/config/FeatureSwitch.scala]: Are we still using these filters?

File: app/uk/gov/hmrc/r2/selfassessmentapi/config/AppContext.scala (hunk 1 above)
Comment (jeremystone) [reply app/uk/gov/hmrc/r2/selfassessmentapi/config/AppContext.scala]: It should be the same as before - `getBoolean` is deprecated and replaced with `getOptional[Boolean]`

File: app/uk/gov/hmrc/r2/selfassessmentapi/config/FeatureSwitch.scala (hunk 2 above)
Comment (jeremystone) [reply app/uk/gov/hmrc/r2/selfassessmentapi/config/FeatureSwitch.scala]: Not sure. There is a separate story to review and remove them. All I did here was play upgrade.

**Pull request merged**